cli/core/test/e2e/config/config_test.go

```diff
@@ -0,0 +1,59 @@
+// Package config_e2e_test provides config command specific E2E test cases
+package config_e2e_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/cluster-api/util"
+
+	"github.com/vmware-tanzu/tanzu-framework/cli/core/test/e2e/framework"
+)
+
+var _ = framework.CLICoreDescribe("[Tests:E2E][Feature:Command-Config]", func() {
+	var (
+		tf *framework.Framework
+	)
+	BeforeEach(func() {
+		tf = framework.NewFramework()
+	})
+	Context("config feature flag operations", func() {
+		When("new config flag set with value", func() {
+			It("should set flag and unset flag successfully", func() {
+				randomFlagName := RandomNameWithE2ETestString()
+				randomFeatureFlagPath := "features.global." + randomFlagName
+				flagVal := "true"
+				err := tf.ConfigSetFeature(randomFeatureFlagPath, flagVal)
+				Expect(err).To(BeNil())
+
+				val, err := tf.ConfigFeatureFlagExists(randomFlagName, flagVal)
+				Expect(err).To(BeNil())
+				Expect(val).To(BeTrue())
+
+				err = tf.ConfigUnsetFeature(randomFeatureFlagPath)
+				Expect(err).To(BeNil())
+
+				val, err = tf.ConfigFeatureFlagNotExists(randomFlagName)
+				Expect(err).To(BeNil())
+				Expect(val).To(BeTrue())
+			})
+		})
+		When("config init called", func() {
+			It("should initialize configuration successfully", func() {
+				err := tf.ConfigInit()
```

Comment:
We need to make sure that `tanzu config init` command performed the things that it should have rather then just checking for error here.

One another way to test this would be to do tests in below sequence:
1. Clean all the config files (we might need a helper function in the framework to cleanup all the configuration that tanzu cli uses)
2. Run `tanzu config init`
3. Verify that `$HOME/.config/tanzu/config.yaml` and `$HOME/.config/tanzu/config-ng.yaml` gets created
4. Run feature-flag set tests
5. Run feature-flag get tests
6. Run `tanzu config init` again
7. Verify that second run of `tanzu config init` doesn't cleanup the existing feature-flag which was already configured
8. Run feature-flag unset tests

Reply:
I have updated the test case, to delete config files, initialize, validate config set, and again initialize and validate config.